cli/azd/cmd/deploy.go

```diff
@@ -211,10 +216,20 @@ func (da *deployAction) Run(ctx context.Context) (*actions.ActionResult, error)
 		Title: "Deploying services (azd deploy)",
 	})
 
+	var stepMessage string
+	progressFilter := func(msg messaging.Message) bool {
+		return msg.Type == project.ProgressMessage
+	}
+	progressSubscription := da.subscriber.Subscribe(ctx, progressFilter, func(msg messaging.Message) {
+		progressMessage := fmt.Sprintf("%s (%s)", stepMessage, msg.Value)
+		da.console.ShowSpinner(ctx, progressMessage, input.Step)
+	})
+	defer progressSubscription.Close()
```

Comment:
In this new model we would no longer need to read progress from individual tasks.  Any progress message published to the topic can automatically handled across project/service lifecycle functions.